src/overlays/gamestates/ovl_file_choose/file_choose.h

```diff
@@ -0,0 +1,219 @@
+    /* 1 */ BTN_SELECT_FILE_2,
+    /* 2 */ BTN_SELECT_FILE_3,
+    /* 3 */ BTN_SELECT_YES,
+    /* 4 */ BTN_SELECT_QUIT
+} SelectMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ BTN_CONFIRM_YES,
+    /* 1 */ BTN_CONFIRM_QUIT
+} ConfirmButtonIndex;
+
+typedef enum {
+    /* 0 */ BTN_ACTION_COPY,
+    /* 1 */ BTN_ACTION_ERASE
+} ActionButtonIndex;
+
+typedef enum {
+    /* 0 */ SETTING_AUDIO,
+    /* 1 */ SETTING_TARGET
+} SettingIndex;
+
+typedef enum {
+    /* 0 */ AUDIO_STEREO,
+    /* 1 */ AUDIO_MONO,
+    /* 2 */ AUDIO_HEADSET,
+    /* 3 */ AUDIO_SURROUND
+} AudioOption;
+
+typedef enum {
+    /* 0 */ CHAR_PAGE_HIRA,
+    /* 1 */ CHAR_PAGE_KATA,
+    /* 2 */ CHAR_PAGE_ENG
+} CharPage;
+
+typedef enum {
+    /* 00 */ KBD_BTN_HIRA,
+    /* 01 */ KBD_BTN_KATA,
+    /* 02 */ KBD_BTN_ENG,
+    /* 03 */ KBD_BTN_BACKSPACE,
+    /* 04 */ KBD_BTN_END,
+    /* 99 */ KBD_BTN_NONE = 99
+} KeyboardButton;
+
+void FileChoose_SetupCopySource(GameState* thisx); 
+void FileChoose_SelectCopySource(GameState* thisx); 
+void FileChoose_SetupCopyDest1(GameState* thisx); 
+void FileChoose_SetupCopyDest2(GameState* thisx); 
+void FileChoose_SelectCopyDest(GameState* thisx); 
+void FileChoose_ExitToCopySource1(GameState* thisx); 
+void FileChoose_ExitToCopySource2(GameState* thisx); 
+void FileChoose_SetupCopyConfirm1(GameState* thisx); 
+void FileChoose_SetupCopyConfirm2(GameState* thisx); 
+void FileChoose_CopyConfirm(GameState* thisx); 
+void FileChoose_ReturnToCopyDest(GameState* thisx); 
+void FileChoose_CopyAnim1(GameState* thisx); 
+void FileChoose_CopyAnim2(GameState* thisx); 
+void FileChoose_CopyAnim3(GameState* thisx); 
+void FileChoose_CopyAnim4(GameState* thisx); 
+void FileChoose_CopyAnim5(GameState* thisx); 
+
+void FileChoose_ExitCopyToMain(GameState* thisx); 
+void FileChoose_SetupEraseSelect(GameState* thisx); 
+void FileChoose_EraseSelect(GameState* thisx); 
+void FileChoose_SetupEraseConfirm1(GameState* thisx); 
+void FileChoose_SetupEraseConfirm2(GameState* thisx); 
+void FileChoose_EraseConfirm(GameState* thisx); 
+void FileChoose_ExitToEraseSelect1(GameState* thisx); 
+void FileChoose_ExitToEraseSelect2(GameState* thisx); 
+void FileChoose_EraseAnim1(GameState* thisx); 
+void FileChoose_EraseAnim2(GameState* thisx); 
+void FileChoose_EraseAnim3(GameState* thisx); 
+void FileChoose_ExitEraseToMain(GameState* thisx); 
+
+void FileChoose_UpdateKeyboardCursor(GameState* thisx);
+void FileChoose_StartNameEntry(GameState* thisx);
+void FileChoose_UpdateOptionsMenu(GameState* thisx);
+void FileChoose_StartOptions(GameState* thisx);
+
+void FileChoose_InitModeDraw(GameState* thisx);
+void FileChoose_ConfigModeDraw(GameState* thisx);
+void FileChoose_SelectModeDraw(GameState* thisx);
+
+void FileChoose_PulsateCursor(GameState* thisx);
+void FileChoose_DrawOptions(GameState* thisx);
+
+void FileChoose_DrawNameEntry(GameState* thisx);
+void FileChoose_DrawCharacter(GraphicsContext* gfxCtx, void* texture, s16 vtx);
+
+extern s16 D_808123F0[];
+
+#endif
```

Comment:
new line

Reply:
theres a newline at the end of the file, if thats what you mean